packages/file-types/ome-tiff/src/OmeTiffLoader.js

```diff
@@ -0,0 +1,93 @@
+import { viv } from '@vitessce/gl';
+import {
+  initializeRasterLayersAndChannels,
+  coordinateTransformationsToMatrix,
+} from '@vitessce/spatial-utils';
+import { AbstractTwoStepLoader, imageOmeTiffSchema, LoaderResult } from '@vitessce/vit-s';
+
+export default class OmeTiffLoader extends AbstractTwoStepLoader {
+  constructor(dataSource, params) {
+    super(dataSource, params);
+    this.optionsSchema = imageOmeTiffSchema;
+  }
+
+  async loadOffsets() {
+    const { offsetsUrl } = this.options || {};
+    if (offsetsUrl) {
+      const res = await fetch(offsetsUrl, (this.requestInit || {}));
+      if (res.ok) {
+        const offsets = await res.json();
+        return offsets;
+      }
+      throw new Error(`OME-TIFF offsets JSON file failed to load: ${res.status} from ${res.url}`);
+    }
+    return null;
+  }
+
+  async load() {
+    const { url, requestInit } = this;
+    const { coordinateTransformations } = this.options || {};
+
+    // Get image name and URL tuples.
+    const urls = [
+      [url, 'image'],
+    ];
+
+    const offsets = await this.loadOffsets();
+    const loader = await viv.loadOmeTiff(url, { offsets, headers: requestInit?.headers });
+    const { Name: imageName, Pixels: { Channels } } = loader.metadata;
+    const channels = Array.isArray(Channels)
+      ? Channels.map((channel, i) => channel.Name || `Channel ${i}`)
+      : [Channels.Name || `Channel ${0}`];
+
+    const image = {
+      name: imageName || 'Image',
+      url,
+      type: 'ome-tiff',
+      ...(coordinateTransformations ? {
+        metadata: {
+          transform: {
+            matrix: coordinateTransformationsToMatrix(coordinateTransformations),
+          },
+        },
+      } : {}),
+    };
+
+    // Add a loaderCreator function for each image layer.
+    const imagesWithLoaderCreators = [
+      {
+        ...image,
+        loaderCreator: async () => ({ ...loader, channels }),
+      },
+    ];
+
+    const usePhysicalSizeScaling = false;
```

Comment:
This makes sense.  Do you see any way forward for this option with these new file types?

Reply:
Just updated this - I am now doing
```
const usePhysicalSizeScaling = (
      PhysicalSizeX
      && PhysicalSizeXUnit
      && PhysicalSizeY
      && PhysicalSizeYUnit
    );
```
does that make sense?

Reply:
I think so.